packages/clerk-js/src/ui/components/SignIn/SignInFactorOne.tsx

```diff
@@ -84,7 +89,20 @@ export function _SignInFactorOne(): JSX.Element {
 
   switch (currentFactor?.strategy) {
     case 'password':
-      return <SignInFactorOnePasswordCard onShowAlternativeMethodsClick={toggleAllStrategies} />;
+      return (
+        <SignInFactorOnePasswordCard
+          onFactorPrepare={() =>
+            setCurrentFactor(prevFactor => {
+              setPrevCurrentFactor(prevFactor);
```

Comment:
Do you think it would be simpler to use one `useState` here with an object containing both factors? Only if it would simplify this part of the code.

Reply:
Do you mean like ? Yes this simplifies the code
```typescript
setFactor((prev)=>({...prev, currFactor}))

setFactor((prev)=>({...prev, prevFactor}))
```